tfx/components/transform/executor_test.py

```diff
@@ -227,51 +243,34 @@ def testCounters(self):
     self.assertMetricsCounterEqual(metrics, 'analyze_paths_count', 1)
 
   def testDoWithCache(self):
-
     # First run that creates cache.
-    output_cache_artifact = _OutputCache()
-    output_cache_artifact.uri = os.path.join(self._output_data_dir, 'CACHE')
-
-    self._output_dict['cache_output_path'] = [output_cache_artifact]
-
     self._exec_properties['module_file'] = self._module_file
     self._transform_executor.Do(self._input_dict, self._output_dict,
                                 self._exec_properties)
     self._verify_transform_outputs()
-    self.assertNotEqual(0,
-                        len(tf.io.gfile.listdir(output_cache_artifact.uri)))
 
     # Second run from cache.
     self._output_data_dir = self._get_output_data_dir('2nd_run')
-    input_cache_artifact = _InputCache()
-    input_cache_artifact.uri = output_cache_artifact.uri
-
-    output_cache_artifact = _OutputCache()
-    output_cache_artifact.uri = os.path.join(self._output_data_dir, 'CACHE')
+    input_cache_artifact = standard_artifacts.TransformCache()
+    input_cache_artifact.uri = self._output_cache_artifact.uri
 
     self._make_base_do_params(self._source_data_dir, self._output_data_dir)
 
-    self._input_dict['cache_input_path'] = [input_cache_artifact]
-    self._output_dict['cache_output_path'] = [output_cache_artifact]
+    self._input_dict[labels.CACHE_INPUT_PATH_LABEL] = [input_cache_artifact]
 
     self._exec_properties['module_file'] = self._module_file
     self._transform_executor.Do(self._input_dict, self._output_dict,
```

Comment:
Consider using self._runPipelineGetMetrics here instead. This way we can verify that this time the num_instances is 0 (see L223 for example), which means that the input cache was used correctly, and it covers the entire input datasets.

Reply:
since the cache is an analyzer cache, wouldnt the transformed paths still contribute to the num_instances count? my understanding is that the analyzer cache just makes it so we don't need to reanalyze already processed paths, but we still are retransforming everything.

Reply:
using metrics, updated test to check that `num_instances` is smaller than initial run `num_instances` by the number of train paths (which should hit cache).